src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -438,40 +450,39 @@ def __init__(self, setup_module_name: str, simulation_config: SimulationConfig,
         )
 
         self.progress_info = SimulationProgressInfo()
-        self.simulation_config = simulation_config
         self._simulation_id = redis_job_id
 
         self._init(redis_job_id)
 
         self._external_events = SimulationExternalEvents(
-            redis_job_id, self.simulation_config, self._status, self.progress_info, self.area)
+            redis_job_id, self._setup.config, self._status, self.progress_info, self.area)
 
         deserialize_events_to_areas(simulation_events, self.area)
 
         validate_const_settings_for_simulation()
 
-    def _init(self, redis_job_id):
+    def _init(self, redis_job_id: str) -> None:
         # has to be called before get_setup():
         global_objects.profiles_handler.activate()
 
-        self.area = self._setup.load_setup_module(self.simulation_config)
+        self.area = self._setup.load_setup_module(self._setup.config)
```

Comment:
Now that the `self._setup` already has access to the simulation_config itself, how about omitting passing it here again?

Reply:
You are right I missed that, will fix it thanks!